### PR TITLE
Use correct direction for anisotropic fission and update numpy version for tests

### LINKED ISSUE #1030: Isotropic fission neutron assumption
Currently OpenMC assumes that the emission of fission neutrons is isotropic, even when the evaluated data says otherwise. It should be straightforward to just use the distribution specified by the evaluation.

### LINKED ISSUE #2190: Last decimal place disagreements causing failed tests with numpy 1.22.0 or later
When testing from a clean copy of the `develop` branch of OpenMC (from the main fork) in debug mode, a few tests fail due to last decimal place or near it disagreements about values in a diff. The `-v` output from `test_lethargy_bin_width` shows one such example
```
test_lethargy_bin_width

    def test_lethargy_bin_width():
        f = openmc.EnergyFilter.from_group_structure('VITAMIN-J-175')
        assert len(f.lethargy_bin_width) == 175
        energy_bins = openmc.mgxs.GROUP_STRUCTURES['VITAMIN-J-175']
        assert f.lethargy_bin_width[0] == math.log10(energy_bins[1]/energy_bins[0])
>       assert f.lethargy_bin_width[-1] == math.log10(energy_bins[-1]/energy_bins[-2])
E       assert 0.05429280308859366 == 0.05429280308859367
E        +  where 0.05429280308859367 = <built-in function log10>((19640000.0 / 17332000.0))
E        +    where <built-in function log10> = math.log10

unit_tests/test_filters.py:257: AssertionError
```
This test and a few others fail for me when I was using `python 3.10.5` and `numpy 1.23.0`. Did some digging with @pshriwise and we determined that with `python 3.8.3` and `numpy 1.21.0` these tests pass, but as soon as `numpy` is increased to or beyond `1.22.0`, the assertions fail.

I was wondering if the appropriate fix would be to allow `assert` to vary with a certain tolerance (say like `1e-15`) or if something else should be done with the environment to make sure they exactly agree.

### PR DESCRIPTION
**Pull request by paulromano**
This PR ensures that fission specifies an anisotropic distribution of fission neutrons, they are emitted in the correct direction. If you look at physics.cpp, you'll see that right now, we sample the correct (possibly anisotropic distribution) but we apply the μ value to a vector pointing at (0, 0, 1) rather than using the incident direction of the neutron. Fixing this resulted in all our regression test results changing, so I figured I'd also make appropriate updates for our tests to use a more recent numpy version (since that also was causing problems with some tests; see #2190). There are also a few random changes to get rid of warnings that were popping up during tests.

Closes #1030
Closes #2190

### CODE REVIEW

**Comment by paulromano** [comment]
It was a rather long ordeal for me to figure out why tests were failing on this PR and I wanted to summarize what I learned.

Updating from `numpy<1.22` as we were using before to the most recent numpy was resulting in `regression_tests/source` randomly failing where the inputs that were generated (based on numpy functions like `logspace`) would sometimes differ in the last digit. There was no rhyme or reason as to the failures in CI -- sometimes they would fail, and sometimes they wouldn't. After a lot of digging, I found out that the wheels for numpy 1.22+ now include [runtime CPU dispatch for SIMD instructions](https://numpy.org/doc/stable/reference/simd/index.html). What this means is that the C-level code in numpy is compiled with multiple versions using different SIMD intrinsics (e.g., SSE, AVX, AVX2, AVX512). Then, at runtime, numpy looks at what CPU you're on and what intrinsics are available and will use the most performant version. This is great, but it means that depending on which set of SIMD instructions are used, you can get slightly different answers in the last digit. Furthermore, GitHub Actions uses [`Standard_DS2_v2` virtual machines](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#cloud-hosts-used-by-github-hosted-runners) from Microsoft Azure, which may run on [various types of Intel Xeons](https://learn.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series). If you get assigned a more recent Xeon, it will likely have AVX512, whereas if you get an older one, it won't. The solution here was to disable AVX512 through the `NPY_CPU_DISABLE_FEATURES` environment variable which now shows up in ci.yml.

Even after figuring out the CPU dispatch issue, I was then seeing that our MPI build configurations were completely hanging on the `deplete_no_transport` test and would just timeout at the 360 minute limit. The issue here turned out to be that updating to the latest Ubuntu pulled in a more recent version of libfabric (used by MPI implementations) which was complaining about calls to the fork() system call (these are a result of the use of the Python multiprocessing module in depletion). I had to add another environment variable (`RDMAV_FORK_SAFE=1`) to fix that one.

**Comment by paulromano** [comment]
I've run a set of >100 criticality benchmarks on this branch compared to the `develop` branch, which gives the following differences. As you can see, everything is basically within stochastic uncertainty:
anisotropic-fission

**Review by pshriwise — APPROVED** [review]
Thanks for tracking down those CI issues @paulromano! Sounds like that required some real digging in NumPy and GitHub actions! 

The criticality benchmarks look good as well. Thanks for running those 💯

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,27 +45,27 @@ jobs:
             omp: n
             mpi: n
           - dagmc: y
-            python-version: '3.10'
+            python-version: "3.10"
```
Comment (pshriwise) [inline .github/workflows/ci.yml:48]: Out of curiosity, why the change from single to double quotes?

Comment (paulromano) [reply .github/workflows/ci.yml:48]: My editor deciding for me that double quotes are better 🤷‍♂️

Comment (pshriwise) [reply .github/workflows/ci.yml:48]: Right on! Just making sure there wasn't a technical difference.
